docs/developer-docs/web-apps/verifiable-credentials/issuer.mdx

```diff
@@ -0,0 +1,164 @@
+
+Internet Identity is used as the identity provider in this tutorial.
+
+## Development of the issuer API
+
+The issuer API consists of four endpoints:
+
+1. [`vc_consent_message`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#1-consent-message): Responds with a message of consent to be displayed to the user to receive their consent on issuing the credentials.
+
+2. [`derivation_origin`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#2-derivation-origin): Returns the URL used to derive the user's principal for the issuer. 
+
+3. [`prepare_credential`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#3-prepare-credential): Check the validity of the request and prepare the actual credential requested.
+
+4. [`get_credential endpoint`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#4-get-credential): Issues the actual credential requested by the user.
+
+### Step 1: Obtain a consent message (`vc_consent_message`)
+
+First, the user must send a consent message to the issuer API endpoint `vc_consent_message` through the identity provider. A consent message is a text shown to the user to confirm that they consent to the issuing of a VC.
+
+An example implementation in Rust can be found below:
+
+```rust
+pub fn get_vc_consent_message_en(
+    credential_spec: &CredentialSpec,
+) -> Result<Icrc21ConsentInfo, Icrc21Error> {
+    match verify_credential_spec(credential_spec) {
+        Ok(SupportedCredentialType::EarlyAdopter(since_year)) => Ok(Icrc21ConsentInfo {
+            consent_message: format!("You became an early adopter in {}.", since_year),
+            language: "en".to_string(),
+        }),
+        Ok(SupportedCredentialType::EventAttendance(event_name)) => Ok(Icrc21ConsentInfo {
+            consent_message: format!("You have attended the event {}.", event_name),
+            language: "en".to_string(),
+        }),
+        Err(err) => Err(Icrc21Error::ConsentMessageUnavailable(Icrc21ErrorInfo {
+            description: format!("Credential spec not supported: {:?}", err),
+        })),
+    }
+}
+```
+
+### Step 2: Implement a derivation origin (`derivation_origin`)
+
+Next, the issuer API endpoint `derivation_origin` is called by the identity provider to obtain a URL to be used as the derivation origin for user's principal.
+
+If the issuer doesn't rely on alternative derivation origins, then it must return the canister's URL: `https://<issuer-canister-id>.icp0.io`.
+
+If the issuer relies on the [alternative derivation origins](/docs/current/references/ii-spec#alternative-frontend-origins) feature, the endpoint must return the same value for the `derivationOrigin` when logging in.
+
+The returned value is subject to verification via .well-known/ii-alternative-origins, as describe in the alternative origins documentation.
```

Comment:
This concept is/was unclear to an implementer (me). I remember thinking: "Should I provide the .well-known/ii-alternative-origins, and if so, where and with what contents?". From my perspective, when I was building an issuer and a RP at the same time it was not clear if the derivation origin information concerned the issuer or the RP. One or two examples and a diagram would go a long way explaining how it all connects.

Reply:
I understand, I saw the same issues in another project.

We will add a section to help devs with local development and troubleshooting common problems.

We are also working on a small change to the [VC Spec](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md) to improve the dev experience when the issuer uses an architecture of multiple canisters (multiple or one backend + one frontend).

I'll keep you posted when we do the change.

Btw, did you see that we published an [npm library](https://github.com/dfinity/verifiable-credentials-sdk/tree/main/js-library) for relying parties? It's still under development and might come with (small) breaking changes, but it should ease the development of the relying party.